BootloaderCorePkg/Stage1B/Stage1B.c

```diff
@@ -627,15 +670,29 @@ ContinueFunc (
   BoardInit (PostTempRamExit);
   AddMeasurePoint (0x2070);
 
-  // Extend External Config Data hash
   if (MEASURED_BOOT_ENABLED() ) {
     if (GetBootMode() != BOOT_ON_S3_RESUME) {
+        // Extend External Config Data hash
         if (Stage1bParam->ConfigDataHashValid == 1) {
           TpmExtendConfigData (Stage1bParam->ConfigDataHash,
             PcdGet8(PcdCompSignHashAlg),
             (UINT8 *) Stage1bParam->CfgDataAddr,
             ((CDATA_BLOB *) Stage1bParam->CfgDataAddr)->UsedLength);
         }
+
+        //Extend Key manifest hash
+        if (Stage1bParam->KeyHashManifestHashValid == 1) {

```

Comment:
Can we rework the flow to align between config data and hash store ?   Right, they are handled differently.

Reply:
They are handled differently. In Config Data I have introduced an TPMLib API where we handle cases as calcluatehash.
Here in hash store -   external hash store memory read from flash is not available. It is getting overwritten.
So, we are trying handle all cases while hash store is verified. 

I could move additional hash checks done to TPMLib.